core/ssz.go

```diff
@@ -0,0 +1,380 @@
+func (b *VersionedSignedBlindedBeaconBlock) sszValFromVersion(version eth2spec.DataVersion) (sszType, error) {
+	switch version {
+	case eth2spec.DataVersionBellatrix:
+		if b.Bellatrix == nil {
+			b.Bellatrix = new(eth2bellatrix.SignedBlindedBeaconBlock)
+		}
+
+		return b.Bellatrix, nil
+	case eth2spec.DataVersionCapella:
+		if b.Capella == nil {
+			b.Capella = new(eth2capella.SignedBlindedBeaconBlock)
+		}
+
+		return b.Capella, nil
+
+	//	if b.Deneb == nil {
+	//		b.Deneb = new(v1deneb.SignedBlindedBeaconBlock)
+	//	}
+	//	return b.Deneb, nil
+	default:
+		return nil, errors.New("invalid version")
+	}
+}
+
+// ================== VersionedBlindedBeaconBlock ===================
+
+// MarshalSSZ ssz marshals the VersionedBlindedBeaconBlock object.
+func (b VersionedBlindedBeaconBlock) MarshalSSZ() ([]byte, error) {
+	resp, err := ssz.MarshalSSZ(b)
+	if err != nil {
+		return nil, errors.Wrap(err, "marshal VersionedSignedBlindedBeaconBlock")
+	}
+
+	return resp, nil
+}
+
+// MarshalSSZTo ssz marshals the VersionedBlindedBeaconBlock object to a target array.
+func (b VersionedBlindedBeaconBlock) MarshalSSZTo(buf []byte) ([]byte, error) {
+	return marshalSSZVersionedTo(buf, b.Version, b.sszValFromVersion)
+}
+
+// UnmarshalSSZ ssz unmarshals the VersionedBlindedBeaconBlock object.
+func (b *VersionedBlindedBeaconBlock) UnmarshalSSZ(buf []byte) error {
+	version, err := unmarshalSSZVersioned(buf, b.sszValFromVersion)
+	if err != nil {
+		return errors.Wrap(err, "unmarshal VersionedSignedBeaconBlock")
+	}
+
+	b.Version = version
+
+	return nil
+}
+
+// SizeSSZ returns the ssz encoded size in bytes for the VersionedBlindedBeaconBlock object.
+func (b VersionedBlindedBeaconBlock) SizeSSZ() int {
+	val, err := b.sszValFromVersion(b.Version)
+	if err != nil {
+		// SSZMarshaller interface doesn't return an error, so we can't either.
+		return 0
+	}
+
+	return sizeSSZVersioned(val)
+}
+
+// sszValFromVersion returns the internal value of the VersionedBlindedBeaconBlock object for a given version.
+func (b *VersionedBlindedBeaconBlock) sszValFromVersion(version eth2spec.DataVersion) (sszType, error) {
+	switch version {
+	case eth2spec.DataVersionBellatrix:
+		if b.Bellatrix == nil {
+			b.Bellatrix = new(eth2bellatrix.BlindedBeaconBlock)
+		}
+
+		return b.Bellatrix, nil
+	case eth2spec.DataVersionCapella:
+		if b.Capella == nil {
+			b.Capella = new(eth2capella.BlindedBeaconBlock)
+		}
+
+		return b.Capella, nil
+	//	if b.Deneb == nil {
+	//		b.Deneb = new(v1deneb.SignedBlindedBeaconBlock)
+	//	}
+	//	return b.Deneb, nil
+	default:
+		return nil, errors.New("invalid version")
+	}
+}
+
+// versionedOffset is the offset of a versioned ssz encoded object.
+const versionedOffset = 8 + 4 // version (uint64) + offset (uint32)
+
+// marshalSSZVersionedTo marshals a versioned object to a target array.
+func marshalSSZVersionedTo(dst []byte, version eth2spec.DataVersion, valFunc func(eth2spec.DataVersion) (sszType, error)) ([]byte, error) {
```

Comment:
why are we passing valFunc instead of passing exact value?

Reply:
because we need more than one value